docs/guides/writing_stubs.rst

```diff
@@ -575,6 +600,14 @@ No::
         ...
     def to_int3(x: str) -> int: pass
 
+Avoid invariant collection types (`list`, `dict`) for function parameters,
+in favor of covariant types like `Mapping` or `Sequence`.
+
+Avoid union return types. See https://github.com/python/mypy/issues/1693
```

Comment:
Probably for a future PR, but it would be nice if the content of https://github.com/python/mypy/issues/1693 and https://github.com/python/mypy/issues/7214 below could be copied into the guides. Having pieces of general typing advice hidden away in mypy issues feels sort of odd.